app/src/main/java/com/example/learn09082021v001/MainActivity.kt

```diff
@@ -37,39 +40,47 @@ class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+        // view config
+        viewMainAndroid.edtName.isEnabled = false
+        viewMainIos.edtName.isEnabled = false
+        viewMainAndroid.btnDEC.visibility = View.GONE
+        viewMainAndroid.btnINC.visibility = View.GONE
+        viewMainIos.btnDEC.visibility = View.GONE
+        viewMainIos.btnINC.visibility = View.GONE
         btnStartReset.setOnClickListener {
-            if (tvViewNameAndroid.text.isEmpty() && tvViewNameIOS.text.isEmpty() &&
-                tvViewScoreAndroid.text.isEmpty() && tvViewScoreIOS.text.isEmpty()
+            if (viewMainAndroid.edtName.text.isEmpty() && viewMainAndroid.tvScore.text.isEmpty() &&
+                viewMainIos.edtName.text.isEmpty() && viewMainIos.tvScore.text.isEmpty()
             ) {
                 val intent = Intent(this, DataInput::class.java)
                 startActivityForResult(intent, 1)
-            } else {
-                clearData()
-            }
+            } else clearData()
         }
     }
-    // Funtion
+
+    private fun upDateDataMain() {
+        viewMainAndroid.edtName.setText(scoreOOP.NameInputandroid)
+        viewMainAndroid.tvScore.text = scoreOOP.ScoreInPutandroid.toString()
+        viewMainIos.edtName.setText(scoreOOP.NameInPutIOS)
+        viewMainIos.tvScore.text = scoreOOP.ScoreInPutIOS.toString()
+    }
+// Funtion
 
     private fun checkDataFull() {
-        if (tvViewNameAndroid.text.isNotEmpty() || tvViewNameIOS.text.isNotEmpty()
-            || tvViewScoreAndroid.text.isNotEmpty() || tvViewScoreIOS.text.isNotEmpty()
+        if (viewMainAndroid.edtName.text.isNotEmpty() || viewMainIos.edtName.text.isNotEmpty()
+            || viewMainAndroid.tvScore.text.isNotEmpty() || viewMainIos.tvScore.text.isNotEmpty()
```

Comment:
this also can shorter. Declare a new function called isEmpty in the custom view
fun isNotEmpty():Boolean {
return edtName.text.isNotEmpty() || text.isNotEmpty()
}

after that, in here can call 
if(viewMainAndroid.isNotEmpty) || viewMainIos.isNotEmpty())